src/Routing/Types.purs

```diff
@@ -0,0 +1,6 @@
+module Routing.Types where
+
+import qualified Data.StrMap as M
+
+data RoutePart = Path String | Query (M.StrMap String)
+type Route = [RoutePart]
```

Comment:
Oops, sorry, I didn't realize you needed to support those cases. Isn't it more common to push the query parameters to the end?